htdocs/compta/facture/class/api_invoices.class.php

```diff
@@ -839,6 +839,118 @@ function settounpaid($id)
     }
 
+   /**
+     * Make credt note from invoice
+     *
+     * @param   int 	$id            Invoice ID
+     * @url POST    {id}/makecreditnote
+     *
+     * @return  array 	An invoice object
+     *
+     * @throws 200
+     * @throws 304
+     * @throws 401
+     * @throws 404
+     * @throws 500
+     */
+    function makecreditnote($id)
+    {
+        if(! DolibarrApiAccess::$user->rights->facture->creer) {
+                throw new RestException(401);
+        }
+		
+	    $this->db->begin();
+		
+        $result = $this->invoice->fetch($id);
+        if( ! $result ) {
+                throw new RestException(404, 'Invoice not found');
+        }
+
+        if( ! DolibarrApi::_checkAccessToResource('facture',$this->invoice->id)) {
+                throw new RestException(401, 'Access not allowed for login '.DolibarrApiAccess::$user->login);
+        }
+
+		$result = $this->invoice->fetch_thirdparty();
+        if( ! $result ) {
+                throw new RestException(404, 'Thirdparty not found');
+        }
+
+		if (! $object->paye)	// protection against multiple submit
+		{
+		   	$this->invoice->fetch_lines();
+				
+			// Boucle sur chaque taux de tva
+			$i=0;
+			foreach($this->invoice->lines as $line)
+			{
+				$amount_ht[$line->tva_tx]+=$line->total_ht;
+				$amount_tva[$line->tva_tx]+=$line->total_tva;
+				$amount_ttc[$line->tva_tx]+=$line->total_ttc;
+				$i++;
+			}
+
+			// Insert one discount by VAT rate category
+			$discount = new DiscountAbsolute($this->db);
+			if ($this->invoice->type == 2)     $discount->description='(CREDIT_NOTE)';
+			elseif ($this->invoice->type == 3) $discount->description='(DEPOSIT)';
+			else {
+				$this->error="CantConvertToReducAnInvoiceOfThisType";
+				return -1;
+			}
+			$discount->tva_tx=abs($this->invoice->total_ttc);
+			$discount->fk_soc=$this->invoice->socid;
+			$discount->fk_facture_source=$this->invoice->id;
+
+			$error=0;
+			foreach($amount_ht as $tva_tx => $xxx)
+			{
+				$discount->amount_ht=abs($amount_ht[$tva_tx]);
+				$discount->amount_tva=abs($amount_tva[$tva_tx]);
+				$discount->amount_ttc=abs($amount_ttc[$tva_tx]);
+				$discount->tva_tx=abs($tva_tx);
+
+				$result=$discount->create(DolibarrApiAccess::$user);
+				if ($result < 0)
+				{
+					$error++;
+					break;
+				}
+			}
+
+			if (! $error)
+			{
+				// Classe facture
+				$result=$this->invoice->set_paid(DolibarrApiAccess::$user);
+				if ($result > 0)
+				{
+					//$mesg='OK'.$discount->id;
+					$this->db->commit();
+				}
+				else
+				{
+					throw new RestException(500, 'Could not set paid');
+					$this->db->rollback();
+				}
+			}
+			else
+			{
+				throw new RestException(500, 'Discount error');
+				$this->db->rollback();
+			}
+		}
+
+        $result = $this->invoice->fetch($id);
+        if( ! $result ) {
+            throw new RestException(404, 'Invoice not found');
+        }
+
+        if( ! DolibarrApi::_checkAccessToResource('facture',$this->invoice->id)) {
```

Comment:
@Rusu-Marian-Dan also here

Reply:
Thanks man

What are you using to spot indent issues?

Thanks
Dan

Reply:
Hi,

Have done the fix and pushed to my fork, do I need to make a new pull request?

Dan

Reply:
![2019-02-11 19_58_31-new _ api create credit note from invoice by rusu-marian-dan pull request 105](https://user-images.githubusercontent.com/3624836/52586321-828f0200-2e37-11e9-95f4-1f2274c39617.png)

Reply:
![travis](https://user-images.githubusercontent.com/3624836/52586566-1e207280-2e38-11e9-83f8-2b2d2205e8ef.gif)

Reply:
Cool
Thanks man
Dan

Reply:
Got it

I think have fixed mine,
There is another issue on Dolibarr/dolibarr/htdocs/core/modules/modTicket.class.php

Do you want me to fix that as well? I do not think is my change but does not seem to be complicated

Sorry for the basic questions, I am new to open source

Thanks
Dan

Reply:
the issue in modticket class has been fixed, you can ignore it.

Reply:
Cheers,
Do I need to do a new pull request?
Or my fix can be used from the original pull request I have made.